viewer/src/views/threejs/post-processing/ssao.ts

```diff
@@ -0,0 +1,354 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import glsl from 'glslify';
+import * as THREE from 'three';
+import { SimplexNoise } from 'three/examples/jsm/math/SimplexNoise';
+import { RenderType } from '../materials';
+import { CadNode } from '../cad/CadNode';
+
+const vertexShaderAntialias = glsl(require('../../../glsl/post-processing/fxaa.vert').default);
+const fragmentShaderAntialias = glsl(require('../../../glsl/post-processing/fxaa.frag').default);
+const passThroughVertexShader = glsl(require('../../../glsl/post-processing/passthrough.vert').default);
+const ssaoShader = glsl(require('../../../glsl/post-processing/ssao.frag').default);
+const ssaoFinalShader = glsl(require('../../../glsl/post-processing/ssao-final.frag').default);
+
+interface Uniforms {
+  [uniform: string]: THREE.IUniform;
+}
+
+function setupRenderingPass(options: {
+  uniforms: Uniforms;
+  vertexShader: string;
+  fragmentShader: string;
+}): THREE.Scene {
+  const scene = new THREE.Scene();
+  const material = new THREE.ShaderMaterial({
+    uniforms: options.uniforms,
+    vertexShader: options.vertexShader,
+    fragmentShader: options.fragmentShader,
+    depthWrite: false
+  });
+  const quad = new THREE.Mesh(new THREE.PlaneBufferGeometry(2, 2), material);
+  quad.frustumCulled = false;
+  scene.add(quad);
+  return scene;
+}
+
+interface SceneInfo {
+  scene: THREE.Scene;
+  uniforms: Uniforms;
+}
+
+function createSSAOFinalScene(diffuseTexture: THREE.Texture, ssaoTexture: THREE.Texture): SceneInfo {
+  const uniforms = {
+    tDiffuse: { value: diffuseTexture },
+    ssaoTexture: { value: ssaoTexture },
+    size: { value: new THREE.Vector2() }
+  };
+  const scene = setupRenderingPass({
+    uniforms,
+    vertexShader: passThroughVertexShader,
+    fragmentShader: ssaoFinalShader
+  });
+  return { scene, uniforms };
+}
+
+function createAntialiasScene(diffuseTexture: THREE.Texture): SceneInfo {
+  const uniforms = {
+    tDiffuse: { value: diffuseTexture },
+    resolution: { value: new THREE.Vector2() }
+  };
+  const scene = setupRenderingPass({
+    uniforms,
+    vertexShader: vertexShaderAntialias,
+    fragmentShader: fragmentShaderAntialias
+  });
+  return { scene, uniforms };
+}
+
+const quadCamera = new THREE.OrthographicCamera(-1, 1, 1, -1, 0, 1);
+
+export interface Pass {
+  render: (
+    renderer: THREE.WebGLRenderer,
+    scene: THREE.Scene,
+    camera: THREE.PerspectiveCamera,
+    pass?: SsaoPassType
+  ) => void;
+  setSize: (width: number, height: number) => void;
```

Comment:
Can this be made implicit and decided by the size of the renderer provided in render()?

Reply:
Done